rts/Sim/Misc/Team.cpp

```diff
@@ -221,10 +221,20 @@ void CTeam::GiveEverythingTo(const unsigned toTeam)
 
 	const auto& teamUnits = unitHandler.GetUnitsByTeam(teamNum);
 
+	// To avoid problems with hitting target's maxUnits limit,
+	// temporarily give all of team's unit limit to target
```

Comment:
The "problems" are part of the point of having a limit. How is the game supposed to make sure GiveEverythingTo can't exceed the target team's limit? It can increase the limit on its own if it wants to.

Reply:
I think I just phrased the comment poorly. Here is what I'm trying to say:

Prior to this PR, if A tries to GiveEverythingTo(B) but B.numUnits == B.maxUnits, the unit transfers would fail.

"To avoid that old problem", we first give all of A's unit limit to B (this is potentially dangerous if some transfers fail).

Then, if some units can't transfer to B (ie specific unit limit caps set by modRules or whatever), we adjust for that at the end.

What's a better way to say that?

Reply:
I think the comment is better now.